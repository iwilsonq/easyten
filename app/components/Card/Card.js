import React from 'react';
import { MaterialCommunityIcons } from '@expo/vector-icons';

import theme from '../../config/theme';

import { ImageWrapper } from '../Image';
import { BodyText, HeaderText } from '../Text';
import { WorkoutDate } from '../Date';
import { InlineStats } from '../Stats';
import {
	CardWrapper,
	CardTitle,
	CardBody,
	WorkoutDetails,
	ActionWrapper
} from './styles';

const Card = ({ title = '1:21:13', body = 'Active Time' }) => {
	return (
		<CardWrapper>
			<CardTitle>{title}</CardTitle>
			<CardBody>{body}</CardBody>
		</CardWrapper>
	);
};

const CardAction = ({ flex }) => {
	return (
		<ActionWrapper style={{ flex }}>
			<MaterialCommunityIcons
				name="chevron-right"
				size={26}
				style={{ color: theme.white }}
			/>
		</ActionWrapper>
	);
};

const WorkoutCard = ({
	title = 'A Title',
	distance = '9.86',
	pace = `7'02"`,
	duration = '01:20:55',
	created = Date.now()
}) => {
	return (
		<CardWrapper style={{ flexDirection: 'row' }}>
			<ImageWrapper flex={2} />
			<WorkoutDetails flex={6}>
				<WorkoutDate created={created} />
				<HeaderText>{title}</HeaderText>
				<InlineStats
					duration={duration}
					distance={distance}
					pace={pace}
				/>
			</WorkoutDetails>
			<CardAction flex={1} />
		</CardWrapper>
	);
};

export { Card, WorkoutCard };
