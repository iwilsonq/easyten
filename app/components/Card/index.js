export { Card, WorkoutCard } from './Card';
