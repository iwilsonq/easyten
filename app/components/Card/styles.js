import g from 'glamorous-native';
import { theme, typography } from '../../config/theme';

const CardWrapper = g.view(
	{
		backgroundColor: theme.cardBg,
		borderRadius: 4,
		padding: 12,
		margin: 10
	},
	({ style }) => ({ ...style })
);

const CardTitle = g.text({
	...typography.header
});

const CardBody = g.text({
	...typography.body
});

const WorkoutDetails = g.view(
	{
		marginHorizontal: 12,
		height: 66
	},
	({ flex }) => ({ flex })
);

const ActionWrapper = g.view(
	{
		flex: 1,
		justifyContent: 'center',
		alignItems: 'center'
	},
	({ flex }) => ({ flex })
);

export { CardTitle, CardBody, CardWrapper, WorkoutDetails, ActionWrapper };
