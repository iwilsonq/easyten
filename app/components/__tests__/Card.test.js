import React from 'react';
import renderer from 'react-test-renderer';
import { Card } from '../Card';

describe('Card', () => {
	const props = {};
	const wrapper = renderer.create(<Card {...props} />).toJSON();

	it('renders properly', () => {
		expect(wrapper).toMatchSnapshot();
	});
});
