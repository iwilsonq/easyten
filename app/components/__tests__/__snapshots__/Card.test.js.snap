// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card renders properly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#32343a",
        "borderRadius": 4,
        "margin": 10,
        "padding": 16,
      },
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#dededf",
          "fontFamily": "montserrat-semibold",
          "fontSize": 18,
          "fontWeight": "600",
          "paddingBottom": 8,
        },
      ]
    }
  >
    1:21:13
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#dededf",
          "fontFamily": "montserrat-regular",
          "fontSize": 14,
        },
      ]
    }
  >
    Active Time
  </Text>
</View>
`;
